File: Subarray with given sum(no negtive numbers).py
def SubArraySum(a,s,n):
    st=a[0]
    j=0
    i=1
    while i<=n:
        while st>s and j<i-1:
            st=st-a[j]
            j=j+1
        if st==s:
            print('subarray from ',j,'to ',i-1)
            return 1
        if i<n:
            st=st+a[i]
        i+=1
    cout<<'not found'
    


if __name__=="__main__":
    n=int(input())
    s=int(input())
    a= [ int(x) for x in input().split()]
    SubArraySum(a,s,n)

File: array/Subarray with given sum(no negtive numbers).cpp
#include<bits/stdc++.h> 
using namespace std; 

int main() 

 {    
   #ifndef ONLINE_JUDGE 
   freopen("input.txt", "r", stdin);  
   freopen("output.txt", "w", stdout); 
   #endif 

   int n,sum;
   cin>>n>>sum;
   vector<int> v(n,0);
   for(auto &x:v)
   {
      cin>>x;
   }   
      int j=0;
       int s=v[0];
     for(int i=1;i<v.size();i++)
     {  cout<<s<<endl;
         while(s>sum) && j<i-1)

         {
            s=s-v[j];
            j++;
         }
         if(s==sum)
            {cout<<"Subarray at "<<j<<" to "<<i-1;
               return 0;}

          if(s<sum)
          {
            s=s+v[i];
            
          }     
     }

cout<<"not found";
}
